pkg/kfapp/ossm/ossm_installer.go

```diff
@@ -78,187 +75,131 @@ func (o *OssmInstaller) Init(_ kftypesv3.ResourceEnum) error {
-		return internalError(err)
-	}
-
-	return nil
-}
-
-func (o *OssmInstaller) Generate(resources kftypesv3.ResourceEnum) error {
-	// TODO sort by Kind as .Apply does
-	if err := o.applyManifests(); err != nil {
+	var rootDir = filepath.Join(feature.BaseOutputDir, o.Namespace, o.Name)
+	if err := copyEmbeddedFiles("templates", rootDir); err != nil {
 		return internalError(errors.WithStack(err))
 	}
 
-	o.PatchODHDashboardConfig(o.Namespace)
-
-	o.onCleanup(
-		o.oauthClientRemoval(),
-		o.ingressVolumesRemoval(),
-		o.externalAuthzProviderRemoval(),
-	)
-
-	return nil
-}
-
-// ExtractHostNameAndPort strips given URL in string from http(s):// prefix and subsequent path,
-// returning host name and port if defined (otherwise defaults to 443).
-//
-// This is useful when getting value from http headers (such as origin).
-// If given string does not start with http(s) prefix it will be returned as is.
-func ExtractHostNameAndPort(s string) (string, string, error) {
-	u, err := url.Parse(s)
-	if err != nil {
-		return "", "", err
-	}
-
-	if u.Scheme != "http" && u.Scheme != "https" {
-		return s, "", nil
-	}
-
-	hostname := u.Hostname()
-
-	port := "443" // default for https
-	if u.Scheme == "http" {
-		port = "80"
-	}
-
-	if u.Port() != "" {
-		port = u.Port()
-		_, err := strconv.Atoi(port)
-		if err != nil {
-			return "", "", errors.New("invalid port number: " + port)
-		}
+	if oauth, err := feature.CreateFeature("control-plane-oauth").
+		For(o.PluginSpec).
+		UsingConfig(o.config).
+		Manifests(
+			path.Join(rootDir, feature.ControlPlaneDir, "base"),
+			path.Join(rootDir, feature.ControlPlaneDir, "oauth"),
+			path.Join(rootDir, feature.ControlPlaneDir, "filters"),
+		).
+		WithResources(
+			feature.SelfSignedCertificate,
+			feature.EnvoyOAuthSecrets,
+		).
+		WithData(feature.ClusterDetails, feature.OAuthConfig).
+		Preconditions(
+			feature.EnsureCRDIsInstalled("operator.authorino.kuadrant.io", "v1beta1", "authorinos"),
+			feature.EnsureServiceMeshInstalled,
+		).
+		OnDelete(
+			feature.RemoveOAuthClient,
+			feature.RemoveTokenVolumes,
+		).Load(); err != nil {
+		return nil
+	} else {
+		o.features = append(o.features, oauth)
 	}
 
-	return hostname, port, nil
-}
-
-func (o *OssmInstaller) createConfigMap(cfgMapName string, data map[string]string) error {
-
-	configMap := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      cfgMapName,
-			Namespace: o.KfConfig.Namespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion: o.tracker.APIVersion,
-					Kind:       o.tracker.Kind,
-					Name:       o.tracker.Name,
-					UID:        o.tracker.UID,
-				},
-			},
-		},
-		Data: data,
+	if cfMaps, err := feature.CreateFeature("shared-config-maps").
+		For(o.PluginSpec).
+		UsingConfig(o.config).
+		WithResources(feature.ConfigMaps).
+		Load(); err != nil {
+		return err
+	} else {
+		o.features = append(o.features, cfMaps)
+	}
+
+	if serviceMesh, err := feature.CreateFeature("enable-service-mesh").
```

Comment:
nitpick but I think `"meshify-kfdef-ns"` or `"meshify-odh-ns` might be more accurate for this feature name

Reply:
Agree, I did improve those names in 0867d1d